SYCL/ESIMD/esimd_test_utils.hpp

```diff
@@ -37,17 +37,11 @@ namespace esimd_test {
 class ESIMDSelector : public device_selector {
   // Require GPU device
   virtual int operator()(const device &device) const {
-    if (const char *dev_filter = getenv("SYCL_DEVICE_FILTER")) {
-      std::string filter_string(dev_filter);
-      if (filter_string.find("gpu") != std::string::npos)
-        return device.is_gpu() ? 1000 : -1;
-      std::cerr
-          << "Supported 'SYCL_DEVICE_FILTER' env var values is 'gpu' and '"
-          << filter_string << "' does not contain such substrings.\n";
-      return -1;
+    if (device.get_backend() == backend::ext_intel_esimd_emulator) {
+      return 1000;
+    } else {
```

Comment:
We run ESIMD tests on emulator *and* Intel GPU device. So
```suggestion
    if (device.get_backend() == backend::ext_intel_esimd_emulator) {
      return 1000;
    } else if (device.is_gpu() && device_vendor_is_intel(device)) {
      return 1000;
    } else {
```

Reply:
Thanks for the info, the proposed change has been applied!

Reply:
Actually, since (1) ESIMD emulator and Intel GPU device can't co-exist and (2) ESIMD emulator appears as Intel GPU device, then the above can be simplified:
```cpp
if (device.is_gpu() && device_vendor_is_intel(device)) {
  return 1000;
} else {
  return 0;
}
```

Reply:
Who enforces that ESIMD and gpu device cannot co-exist?
As far as I know, nobody does. If I unset SYCL_DEVICE_FILTER in my environment and run sycl-ls then among the devices I see Intel UHD Graphics and ESIMD emulator together and the tests compile and run fine.

Reply:
Also, is the function device_vendor_is_intel above meant as an example or is it actually implemented?
If its an example, any clues as to to what funtion we can use to check that the device is of Intel?

Reply:
@kbobrovs ,
We decided to allow the user to set SYCL_EVICE_FILTER=esimd_emulator:gpu,level_zero:gpu.
In some cases, users mix SYCL kernels with ESIMD kernels, and they want to run them on different devices by setting the SYCL_DEVICE_FILTER appropriately.
Lorenc's work in https://github.com/intel/llvm/pull/6870 will give a lowest priority for esimd_emulator. So, the user can use a custom filter to override this default behavior by giving a high heuristic score for the esimd_emulator.
This way they use both regular GPU with esimd_emulator:gpu.

Reply:
I changed the file esimd_test_utils.hpp so that the esimd_emulator gives a slightly lower score to ordinary gpu in order not to compete with ESIMD when both of them are present, which based on Byoungro's comment above, is a valid assumption to make.